### PR TITLE
feat(jtk): fix boards get --extended filter name rendering

### LINKED ISSUE #291: jtk boards: fix get --extended filter name and column names
Parent: #230

### `jtk boards get <id> --extended`
**Current:** filter name is blank; column config names differ from live data.
**Expected:**
```
23  MON board
Type: scrum   Project: MON (Platform Development)
Filter: board filter for MON board (id: 10084)
Column config: Backlog, Ready for Development, In Development, In Code Review, Ready for QA, Ready for Deployment, Deployed
```

`Filter:` line should show the filter's name and ID. If the board has no associated filter, omit the line (don't render a blank value).

See #230 for full spec.

### PR DESCRIPTION
**Pull request by rianjs**
## Summary
- Fixes blank filter name in `jtk boards get --extended` output
- Before: `Filter:  (id: 10084)` (leading space when name is empty)
- After: `Filter: id: 10084` (clean fallback when name is empty)
- When name is present: `Filter: board filter for MON board (id: 10084)` (unchanged)
- Column config rendering verified correct — uses API column names directly

Closes #291

## Test plan
- [x] Presenter test for empty filter name → `id: 10084` format
- [x] Presenter test for named filter → `name (id: N)` format (existing, unchanged)
- [x] Presenter test for nil config → `-` (existing, unchanged)
- [x] Command-level test for empty filter name with exact output assertion
- [x] Command-level test verifies column names render exactly from API fixture
- [x] `make build && make test && make lint` all pass

### CODE REVIEW

**Comment by rianjs** [comment]
**Blocker**
- None.

**Major**
- None.

**Minor**
- None.

**Nit**
- None.

**Comment by rianjs** [comment]
## Test coverage gaps

**Missing: `formatFilterRef` unit tests**
The new helper `formatFilterRef` is tested only indirectly through `PresentDetail`. Three of its four branches lack direct coverage:
- `f.ID == "" && f.Name == ""` (both empty) — returns `"-"` but never directly asserted
- `f.ID != "" && f.Name != ""` (both present, happy path) — only covered implicitly via the pre-existing `TestPresentBoardDetail_Extended`
- `f.ID != "" && f.Name == ""` (the bug fix branch) — covered by the new test

A small table-driven test for `formatFilterRef` directly would lock all four branches unambiguously and be immune to section-index drift.

**Missing: projection path for empty filter name**
`PresentDetailProjection` calls `formatFilterRef` at line 121, but there is no test for the empty-name case on the projection path. `TestPresentBoardDetailProjection_ContainsAllSpecHeaders` passes a non-empty filter name only. The bug being fixed (blank filter renders as empty string) can silently regress on the `--fields` output path.

**Missing: empty ID + empty name edge case**
`formatFilterRef` returns `"-"` when `ID == ""`, regardless of `Name`. No test passes `BoardFilter{ID: "", Name: "something"}` to confirm the ID-gate takes full precedence. Minor, but worth one line.

**No gaps in the integration test (`TestRunGet_Extended_EmptyFilterName`)**
The cmd-layer test correctly exercises the HTTP roundtrip and asserts both the positive form (`"Filter: id: 10084"`) and the negative guard against a leading space. That part is solid.

File: tools/jtk/internal/present/board_test.go [hunk 1]
```diff
@@ -116,6 +116,56 @@ func TestPresentBoardDetail_Extended(t *testing.T) {
...
+				{Name: "In Development"},
+			},
+		},
+	}
+	model := BoardPresenter{}.PresentDetail(board, config, true)
+
+	filterLine := model.Sections[2].(*present.MessageSection)
+	if filterLine.Message != "Filter: id: 10084" {
+		t.Errorf("empty filter name: expected 'Filter: id: 10084', got %q", filterLine.Message)
+	}
```
Comment (monit-reviewer) [inline tools/jtk/internal/present/board_test.go:162]: 🔵 **Low** (harness-engineering:harness-architecture-reviewer): TestPresentBoardDetail_Extended_EmptyFilterName accesses model.Sections[2] and model.Sections[3] by hard-coded index. If PresentDetail ever reorders or adds sections, this test will either panic or silently validate the wrong section. Prefer scanning by label/type, consistent with how TestPresentBoardDetailProjection_EmptyFilterName iterates fields.

*Reply to this thread when addressed.*

**Review by monit-reviewer — APPROVED** [review]
## Automated PR Review



**Reviewed commit:** `534ce64`

> **Approved** with 1 non-blocking suggestion below. Address at your discretion.

### Summary

| Reviewer | Findings |
|----------|----------|
| harness-engineering:harness-architecture-reviewer | 1 |

<details>
<summary><strong>harness-engineering:harness-architecture-reviewer (1 findings)</strong></summary>

**💡 Suggestion** - `tools/jtk/internal/present/board_test.go:162`
> TestPresentBoardDetail_Extended_EmptyFilterName accesses model.Sections[2] and model.Sections[3] by hard-coded index. If PresentDetail ever reorders or adds sections, this test will either panic or silently validate the wrong section. Prefer scanning by label/type, consistent with how TestPresentBoardDetailProjection_EmptyFilterName iterates fields.

</details>

*2 info-level observations excluded. Run with `--verbose` to include.*

*2 PR discussion threads considered.*

---
<details>
<summary>Completed in 1m 00s | $0.42</summary>

| Field | Value |
|---|---|
| Reviewers | hybrid-synthesis, harness-engineering:harness-architecture-reviewer, harness-engineering:harness-enforcement-reviewer, harness-engineering:harness-knowledge-reviewer, harness-engineering:harness-self-documenting-code-reviewer, security:security-code-auditor |
| Engine | claude |
| Reviewed by | [pr-review-daemon](https://github.com/SignalFT/pr-review-daemon) · [monit-pr-reviewer](https://github.com/apps/monit-pr-reviewer) |
| Duration | 1m 00s wall · 2m 18s compute (Reviewers: 32s · Synthesis: 26s) |
| Cost | $0.42 |
| Tokens | 179.6k in / 8.3k out |
| Turns | 14 |

</details>

File: tools/jtk/internal/present/board_test.go (hunk 1 above)
Comment (rianjs) [reply tools/jtk/internal/present/board_test.go:162]: Consistent with the existing TestPresentBoardDetail_Extended test pattern which also uses hard-coded section indices. The section order is part of the output contract — if it changes, tests should fail.
